scrapy/pipelines/images.py

```diff
@@ -126,16 +126,37 @@ def get_images(self, response, request, info):
         if width < self.min_width or height < self.min_height:
             raise ImageException("Image too small (%dx%d < %dx%d)" %
                                  (width, height, self.min_width, self.min_height))
+        
+        def _warn():
+            from scrapy.exceptions import ScrapyDeprecationWarning
+            import warnings
+            warnings.warn('ImagesPipeline.convert_image() method overriden in a deprecated way, '
+                          'overriden method does not accept response_body argument.',
+                          category=ScrapyDeprecationWarning, stacklevel=1)
 
-        image, buf = self.convert_image(orig_image)
+        if 'response_body' not in get_func_args(self.convert_image):
```

Comment:
I wonder if we should call `get_func_args` only once and cache its response, instead of calling it every time we need to check its result. `self.convert_image` is not likely to change at run time.